irohad/network/impl/block_loader_service.cpp

```diff
@@ -22,16 +22,20 @@ using namespace iroha;
 using namespace iroha::ametsuchi;
 using namespace iroha::network;
 
-BlockLoaderService::BlockLoaderService(std::shared_ptr<BlockQuery> storage)
-    : storage_(std::move(storage)) {
+BlockLoaderService::BlockLoaderService(
+    std::shared_ptr<BlockQueryFactory> block_query_factory)
+    : block_query_factory_(block_query_factory) {
   log_ = logger::log("BlockLoaderService");
 }
 
 grpc::Status BlockLoaderService::retrieveBlocks(
     ::grpc::ServerContext *context,
     const proto::BlocksRequest *request,
     ::grpc::ServerWriter<::iroha::protocol::Block> *writer) {
-  auto blocks = storage_->getBlocksFrom(request->height());
+  auto blocks = block_query_factory_->createBlockQuery() |
+      [request](const auto &block_query) {
```

Comment:
Wouldn't it be a little more safe, if you capture height in lambda instead of pointer? For example, 
`[height = request->height()](const auto...)`